raiden-dapp/src/components/account/backup-state/DownloadStateDialog.vue

```diff
@@ -52,12 +52,17 @@ export default class DownloadStateDialog extends Mixins(NavigationMixin) {
 
     downloadLink.click();
     this.revokeDownloadURL(stateFileURL, downloadLink);
+    this.cancel();
   }
 
   /* istanbul ignore next */
   async getStateFileURL(filename: string): Promise<string> {
-    const state = await this.$raiden.stopAndGetDatabaseDump();
-    const stateJSON = JSON.stringify(state);
+    let stateJSON = '[';
+    let count = 0;
+    for await (const row of this.$raiden.getDatabaseDump()) {
+      stateJSON += (count++ ? ',\n' : '\n') + JSON.stringify(row);
```

Comment:
Same case as the comment above. Given that this is used more than once, even if delegating everything to `stringify` won't work, it maybe interesting to create a helper function to avoid the code repetition.

Reply:
Unfortunately, I didn't find a simple solution to stream a programatically generated  `File` to a URL without collecting it in a string first, but at least the dApp isn't expected to contain that much entriesnas the cli. It's not very useful to try to split and reuse this code with cli, since this one collects in a string (`+=`) and the other stream to a writable object (`response.write`)